pkg/nsx/services/vpc/vpc.go

```diff
@@ -980,50 +981,60 @@ func (s *VPCService) GetGatewayConnectionTypeFromConnectionPath(connectionPath s
 	return parts[2], nil
 }
 
-func (s *VPCService) ValidateGatewayConnectionStatus(nc *v1alpha1.VPCNetworkConfiguration) (bool, string, error) {
-	var connectionPaths []string // i.e. gateway connection paths
-	var cursor *string
+func (s *VPCService) ValidateConnectionStatus(nc *v1alpha1.VPCNetworkConfiguration) (*common.VPCConnectionStatus, error) {
 	org, project, err := nsxProjectPathToId(nc.Spec.NSXProject)
 	if err != nil {
 		log.Error(err, "Failed to parse NSX project in NetworkConfig", "ProjectPath", nc.Spec.NSXProject)
-		return false, "", err
+		return nil, err
 	}
-	pageSize := int64(1000)
+
+	status := &common.VPCConnectionStatus{
+		GatewayConnectionReady:  false,
+		GatewayConnectionReason: common.ReasonGatewayConnectionNotSet,
+		ServiceClusterReady:     false,
+		ServiceClusterReason:    common.ReasonServiceClusterNotSet,
+	}
+	// If ServiceGateway is not enable, both gateway connection and service cluster are not ready
+	profile, err := s.GetVpcConnectivityProfile(nc, nc.Spec.VPCConnectivityProfile)
+	if err != nil {
+		log.Error(err, "Failed to get VPCConnectivityProfile", "path", nc.Spec.VPCConnectivityProfile)
+		return nil, err
+	}
+	if profile.ServiceGateway == nil || profile.ServiceGateway.Enable == nil || !(*profile.ServiceGateway.Enable) {
+		return status, nil
+	}
+	// For DTGW, in day0, even if vlan connection is set in TGW, service cluster might not be configured.
+	// Thus we check the service cluster path in connectivity profile
+	if len(profile.ServiceGateway.ServiceClusterPaths) > 0 {
```

Comment:
shall we also check service_gateway enabled and service cluster path configured, then set ServiceClusterReady?

Reply:
ignore this, you already check if service gateway is enabled before.